Code/Framework/AzCore/AzCore/Math/Aabb.h

```diff
@@ -51,7 +55,7 @@ namespace AZ
         //! Creates an AABB which contains the specified OBB.
         static Aabb CreateFromObb(const Obb& obb);
 
-        Aabb() = default;
+        AZ_DEPRECATED(Aabb() = default, "The Aabb Default Constructor has been deprecated. Please use Aabb::CreateUninitialized() instead.");
```

Comment:
The default constructor for these classes can't actually be deprecated.

That would break all existing code that does
```
AZ::Aabb varName;
AZ::Color varName;
AZ::Vector2 varName;
etc...
```


My suggestion on how to get this to work is have the default constructor [zero initialize](https://en.cppreference.com/w/cpp/language/zero_initialization) the members of the math types and then provide a constructor that does not.

First we can add a tag type to default initialize the math types
```c++
// This code can be put in AzCore/Math/Internal/MathTypes.h

// Tag type used to allow default initialization to be performed for math types
// https://en.cppreference.com/w/cpp/language/default_initialization
// Default initialize will initialize objects with automatic storage initialization to indeterminate reviews
namespace AZ::Math
{
    struct default_initialize_t{};
    inline constexpr default_initialize_t default_initialize{};
}
```

For example the Aabb code could initialize it's Vector3 members as follows.
```c++
// The default constructor for Vector3 will zero initialize the x, y, z membes
Aabb() = default;

Aabb(AZ::Math::default_initialize_t)
    : m_min(AZ::Math::default_initialize)
    , m_max(AZ::Math::default_initialize)
{
}
```

The types of Vector2, Vector3 and Vector4 work with SimdTypes directly that are default initialized to indeterminate values.

So we should update those types with actual default constructor implementations that zero initializes the Simd member and a default_initialize constructor that default initializes the Simd member

For example here is how the Vector3 class can be updated
```c++
// curly braces {} - zero initializes the Simd member 
AZ::Vector3()
    : m_value{} 
{
}

// No initialization of the Simd member takes place here
// It is left in an indeterminate state
AZ::Vector3(AZ::Math::default_initialize)
{
}
```

Reply:
See comment below, but we can get way with aggregate initialization and therefore don't need to change the constructors at all.